Doc: Add known issue 9.0.7: Agent stuck on failed upgrade

karenzone opened the pull request:
Closes: https://github.com/elastic/docs-content/issues/2978

Adds known issue to docs: "Failed upgrades leave agent stuck until restart"

karenzone commented:
To Do: Think through forwardporting strategy with fix coming so soon. 
_"The fix will be included in versions 9.1.4, 8.19.4, 9.0.8, and 8.18.8."_

colleenmcginnis commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+
+**Applies to: {{agent}} 8.18.7, 9.0.7** 
+
+On September 17, 2025, a known issue was discovered that can cause {{agent}} upgrades to get stuck if an upgrade attempt fails early. This happens because the coordinator’s overrideState remains set, leaving the agent in a state that appears to be upgrading.
```
I find it easier to read when references to code are formatted differently than the rest of the text. 

```suggestion
On September 17, 2025, a known issue was discovered that can cause {{agent}} upgrades to get stuck if an upgrade attempt fails early. This happens because the coordinator’s `overrideState` remains set, leaving the agent in a state that appears to be upgrading.
```

colleenmcginnis commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+
+**Conditions**
+
+This issue is triggered if the upgrade fails during one of the early checks inside Coordinator.Upgrade, for example:
```
```suggestion
This issue is triggered if the upgrade fails during one of the early checks inside `Coordinator.Upgrade`, for example:
```

colleenmcginnis commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+
+- {{fleet}} shows the upgrade action in progress, even though the upgrade remains stuck
+- No further upgrade attempts succeed 
+- Elastic-agent status shows an override state indicating upgrade 
```
Should this be "Elastic Agent" or "elastic-agent" instead?

```suggestion
- Elastic Agent status shows an override state indicating upgrade 
```

colleenmcginnis commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+
+**Workaround**
+
+Restart the {{agent}} to clear the coordinator’s overrideState and allow new upgrade attempts to proceed.
```
```suggestion
Restart the {{agent}} to clear the coordinator’s `overrideState` and allow new upgrade attempts to proceed.
```

colleenmcginnis commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+**Resolution**
+This issue was fixed in [#9992](https://github.com/elastic/elastic-agent/pull/9992), which ensures that the coordinator clears its override state whenever an early failure occurs.
+
+The fix will be included in versions 9.1.4, 8.19.4, 9.0.8, and 8.18.8.
```
It looks like 9.1.4 and 8.19.4 are already out, but 9.0.8 and 8.18.8 are not yet released. Any concerns with listing future versions specifically? (I know when we were talking about `applies_to` tags, there was concern about committing to a change making it into a specific future version.) 

```suggestion
The fix is included in versions 9.1.4 and 8.19.4, and planned for versions 9.0.8 and 8.18.8.
```

(I might be overthinking -- I'll leave it to you to decide!)

colleenmcginnis reviewed (approved):
Some minor suggestions ✨

karenzone commented on docs/release-notes/known-issues.md:
```diff
@@ -23,6 +23,37 @@ Known issues are significant defects or limitations that may impact your impleme
+
+- The agent is not upgradeable
+- Capabilities check denies the upgrade
+- Most commonly: When {{agent}} is tamper-protected and Endpoint returns an error during action proxying, for example, because the upgrade action signature is invalid, missing, or fails verification. This causes the coordinator’s override state to be stuck.
```
```suggestion
- When {agent} is tamper-protected, Endpoint must validate that the upgrade action was correctly signed by Kibana to allow the upgrade. If the signature is missing, invalid, or the connection between {agent} and Endpoint was interrupted, the validation fails. This causes the agent coordinator's override state to become stuck until the agent is restarted.
```

karenzone commented:
I added review changes from the [8.18.7 known issue PR ](https://github.com/elastic/ingest-docs/pull/1856)here, and added review changes from this PR to the 8.18.7 PR.  ✅
